### PR TITLE
NCIPPE-135 | Update CRC/Provider

### PR DESCRIPTION
**Pull request by akavalloor**
1. Included logic to send email notification to Patient when the
Provider or CRC is replaced
2. Included logic to send system notification to Patient when their
Provider or CRC is replaced.

### CODE REVIEW

File: src/main/resources/EmailService.properties [hunk 1]
```diff
@@ -56,3 +56,13 @@ email.joining.signature=<p>If you have questions, you can reply to this email or
...
...hemas-microsoft-com:vml" xmlns:w="urn:schemas-microsoft-com:office:word" data-route="true" href="https://${HOSTNAME}/dashboard" style="height:36px;v-text-anchor:middle;width:250px;" arcsize="5%" strokecolor="#1976d2" fillcolor="#1976d2"><w:anchorlock/><center style="color:#ffffff;font-family:Helvetica, Arial,sans-serif;font-size:14px;">View your account settings</center></v:roundrect><![endif]--><a style="background-color: #1976d2; border: 1px solid #1976d2; border-radius: 4px; color: #ffffff; display: inline-block; font-family: Helvetica, Arial,sans-serif; font-size: 16px; line-height: 40px; text-align: center; text-decoration: none; width: 300px; -webkit-text-size-adjust: none; mso-hide: all;" href="https://${HOSTNAME}/dashboard" data-route="true">View your account settings</a></div></td></tr></tbody></table><p>&nbsp;</p><p>Button not working? Copy and paste this link in your browser: <a href="http://${HOSTNAME}/dashboard/participant/%{PatientId}">http://${HOSTNAME}/dashboard/participant/%{PatientId}</a></p><p>Thank you for participating in the Cancer Moonshot\u2120 Biobank!</p><p>- Cancer Moonshot Biobank Staff</p><p>Questions? Reply to this email or call us at 202-000-0000.</p>
```
Comment (debosarkar) [inline src/main/resources/EmailService.properties]: Need to move the signature down to the signature field

File: src/main/resources/EmailService.properties [hunk 2]
```diff
@@ -56,3 +56,13 @@ email.joining.signature=<p>If you have questions, you can reply to this email or
...
```
Comment (debosarkar) [inline src/main/resources/EmailService.properties]: Need to move the signature down to the signature field

File: src/main/java/gov/nci/ppe/services/impl/EmailLogServiceImpl.java [hunk 3]
```diff
@@ -301,5 +301,40 @@ public String sendEmailToAdminAfterFileUpload(Participant participant, String re
...
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String sendEmailToPatientWhenProviderChanges(String recipientEmail, String patientFirstName, String patientId) {
+		String replaceStringWith[] = { patientFirstName , patientId };
+		String replaceThisString[] = { "%{SalutationFirstName}", "%{PatientId}" };
+
+		String htmlBody = emailServiceConfig.getEmailPatientWhenProvidersAreReplacedHtmlBody();
+		String subject = emailServiceConfig.getEmailPatientWhenProvidersAreReplacedSubject();
+		String updatedHtmlBody = StringUtils.replaceEach(htmlBody, replaceThisString, replaceStringWith);
```
Comment (debosarkar) [inline src/main/java/gov/nci/ppe/services/impl/EmailLogServiceImpl.java]: Append the signature

File: src/main/java/gov/nci/ppe/services/impl/EmailLogServiceImpl.java [hunk 4]
```diff
@@ -301,5 +301,40 @@ public String sendEmailToAdminAfterFileUpload(Participant participant, String re
...
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String sendEmailToPatientWhenCRCChanges(String recipientEmail, String patientirstName, String patientId) {
+		String replaceStringWith[] = { patientirstName , patientId };
+		String replaceThisString[] = { "%{SalutationFirstName}", "%{PatientId}" };
+
+		String htmlBody = emailServiceConfig.getEmailPatientWhenCRCIsReplacedHtmlBody();
+		String subject = emailServiceConfig.getEmailPatientWhenCRCIsReplacedSubject();
+		String updatedHtmlBody = StringUtils.replaceEach(htmlBody, replaceThisString, replaceStringWith);
```
Comment (debosarkar) [inline src/main/java/gov/nci/ppe/services/impl/EmailLogServiceImpl.java]: append the signature

File: src/main/resources/EmailService.properties (hunk 1 above)
Comment (akavalloor) [reply src/main/resources/EmailService.properties]: Changes implemented

File: src/main/resources/EmailService.properties (hunk 2 above)
Comment (akavalloor) [reply src/main/resources/EmailService.properties]: Changes implemented

File: src/main/java/gov/nci/ppe/services/impl/EmailLogServiceImpl.java (hunk 3 above)
Comment (akavalloor) [reply src/main/java/gov/nci/ppe/services/impl/EmailLogServiceImpl.java]: Changes implemented

File: src/main/java/gov/nci/ppe/services/impl/EmailLogServiceImpl.java (hunk 4 above)
Comment (akavalloor) [reply src/main/java/gov/nci/ppe/services/impl/EmailLogServiceImpl.java]: Changes implemented

File: src/main/resources/EmailService.properties [hunk 5]
```diff
@@ -56,3 +56,16 @@ email.joining.signature=<p>If you have questions, you can reply to this email or
...
+
+#Common Signature #2
+email.thankyou.signature=<p>Thank you for participating in the Cancer Moonshot\u2120 Biobank!</p><p>- Cancer Moonshot Biobank Staff</p><p>Questions? Reply to this email or call us at 202-000-0000.</p>
```
Comment (debosarkar) [inline src/main/resources/EmailService.properties]: The common signature is 
<p>- Cancer Moonshot Biobank Staff</p><p>Questions? Reply to this email or call us at 202-000-0000.</p>

There are three thak you lines which we'll have to deal with separately

**Pull request merged**